### ISSUE TITLE
UI fails to load job when there is @ in job name in nomad 1.3.0

### ISSUE THREAD

**Issue by Davasny** [hashicorp/nomad#12986]
### Nomad version

```
Nomad v1.3.0 (52e95d64113e01be05d585d8b4c07f6f19efebbc)
```

### Operating system and Environment details
```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

### Issue

Nomad UI returns 404 when there is `@` sign in job name

![image](https://user-images.githubusercontent.com/25726866/168263124-ac9b9fa4-979d-4438-ab2d-e7523b786bf7.png)

### Reproduction steps

Run nomad job with `@` in name

#### Expected Result

UI should show normal job status

#### Actual Result

404

### Job file (if appropriate)

```
job "@hello_world" {

  datacenters = ["dc1"]
  type = "service"

  group "app" {
    count = 2

    network {
      port "http" {
        to = 8000
      }
    }


    service {
      name = "webapp"
      tags = ["urlprefix-/", "traefik.enable=true", "traefik.http.routers.http.rule=Path(`/`)"]
      port = "http"

      check {
        name     = "alive"
        type     = "http"
        path     = "/"
        interval = "10s"
        timeout  = "2s"
      }
    }


    restart {
      attempts = 2
      interval = "30m"
      delay = "15s"
      mode = "fail"
    }

    task "server" {
      driver = "docker"

      config {
        image = "mnomitch/hello_world_server"
        ports = ["http"]
      }

      env {
        MESSAGE = "Hello World!"
      }
    }
  }
}

```

### Nomad Client logs (if appropriate)

UI sends request with namespace set to job name. I think there is some code that extracts namespace from URL by spliting url by `@` sign

```
Request URL: https://[redacted]/v1/job/?namespace=hello_world
```

**Comment by philrenaud** [comment]
Hey @Davasny; thanks for opening this issue. The UI will now handle these links correctly per #13012 .

Builds are running now, but once complete, you should be able to find an artifact to test against [here](https://github.com/hashicorp/nomad/actions/runs/2321802979); this change will be reflected in an upcoming release.

Thanks for reporting the bug!

**Comment by Davasny** [comment]
Hi @philrenaud, I can't information about this fix in 1.3.1 release. Is the code there, but the release info is missing, or it wasn't merged to 1.3.1?

**Comment by jrasell** [comment]
Hi @Davasny; v1.3.1 was a security release only therefore did not include this fix. This fix will be included within the next general release which should be v1.3.2.
